pkg/operator/controllers/checker/unsupportedconditionchecker.go

```diff
@@ -0,0 +1,146 @@
+	"github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
+	"github.com/sirupsen/logrus"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/Azure/ARO-RP/pkg/operator"
+)
+
+const (
+	consoleBannerName   = "aro-sre"
+	clusterOperatorName = "aro"
+	reasonAsExpected    = "AsExpected"
+
+	WORKER_NODE_MINIMUM_COUNT = 3
+
+	OperatorUpgradableFalse = configv1.ConditionFalse
+	OperatorUpgradableTrue  = configv1.ConditionTrue
+)
+
+type UnsupportedConditionChecker struct {
+	log *logrus.Entry
+
+	kubernetescli kubernetes.Interface
+	consolecli    consoleclient.Interface
+	configcli     configclient.Interface
+
+	role string
+}
+
+func NewUnsupportedConditionChecker(log *logrus.Entry, kubernetescli kubernetes.Interface, consolecli consoleclient.Interface, configcli configclient.Interface, role string) *UnsupportedConditionChecker {
+	return &UnsupportedConditionChecker{
+		log:           log,
+		kubernetescli: kubernetescli,
+		consolecli:    consolecli,
+		configcli:     configcli,
+		role:          role,
+	}
+}
+
+func (ucc *UnsupportedConditionChecker) Name() string {
+	return "UnsupportedConditionChecker"
+}
+
+func (ucc *UnsupportedConditionChecker) Check(ctx context.Context) error {
+	if ucc.role != operator.RoleMaster {
+		return nil
+	}
+
+	return ucc.checkWorkerNodeCount(ctx)
+}
+
+func (ucc *UnsupportedConditionChecker) checkWorkerNodeCount(ctx context.Context) error {
+	nodes, err := ucc.kubernetescli.CoreV1().Nodes().List(ctx, metav1.ListOptions{
+		LabelSelector: "node-role.kubernetes.io/worker",
+	})
+	if err != nil {
+		return err
+	}
+
+	if len(nodes.Items) < WORKER_NODE_MINIMUM_COUNT {
+		err = ucc.setClusterOperatorStatus(ctx, OperatorUpgradableFalse)
+		if err != nil {
+			return err
+		}
+
+		return ucc.addConsoleBanner(ctx, fmt.Sprintf("Unsupported Cluster State: There need to be at least %d worker nodes running", WORKER_NODE_MINIMUM_COUNT))
+	}
+
+	err = ucc.setClusterOperatorStatus(ctx, OperatorUpgradableTrue)
+	if err != nil {
+		return err
+	}
+
+	return ucc.consolecli.ConsoleV1().ConsoleNotifications().Delete(ctx, consoleBannerName, metav1.DeleteOptions{})
+}
+
+func (ucc *UnsupportedConditionChecker) setClusterOperatorStatus(ctx context.Context, upgradeableStatus configv1.ConditionStatus) error {
+	co, err := ucc.configcli.ConfigV1().ClusterOperators().Get(ctx, clusterOperatorName, metav1.GetOptions{})
+	if err != nil {
+		ucc.log.Errorf("Failed to get the %q operator: %q", clusterOperatorName, err.Error())
+		return err
+	}
+
+	condition := configv1.ClusterOperatorStatusCondition{
+		Type:               configv1.OperatorUpgradeable,
+		Status:             upgradeableStatus,
+		LastTransitionTime: metav1.Now(),
+		Reason:             reasonAsExpected,
+	}
+
+	v1helpers.SetStatusCondition(&co.Status.Conditions, condition)
+	ucc.log.Infof("Set the %q operator upgradeable status to %q", clusterOperatorName, upgradeableStatus)
```

Comment:
nit: do we need this log message? All logs go through the Gateway, so any traffic we can omit is a good thing :) Will this message enable us to take action on the cluster?

Reply:
@SudoBrendan - Thanks for your review.  The reason for having the log message is  that we would like to provide with more details if a customers queries us as to why they received message X on date Y, we are able to look up in the logs and see what message was in the banner and for how long it was there.